### PR TITLE
feat: create the test/nmea2000_actisense_decode_raw CLI tool

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

**Review by wvmcastro — COMMENTED** [review]
This is for counting how many bytes are received within "complete packages" ? Why is that useful?

**Comment by doudou** [comment]
> This is for counting how many bytes are received within "complete packages" ? Why is that useful?

It does more. Whenever there are bytes thrown away, it gives the diagnostic and the offset. This allows to look into the stream and check whether the issue is with our driver or with the stream itself. I used it to verify the source of the temperature sensor issue we were having.

The code was there, so I thought it would be good to commit. I can also throw it away, for me tanto faz

**Comment by wvmcastro** [comment]
> > This is for counting how many bytes are received within "complete packages" ? Why is that useful?
> 
> It does more. Whenever there are bytes thrown away, it gives the diagnostic and the offset. This allows to look into the stream and check whether the issue is with our driver or with the stream itself. I used it to verify the source of the temperature sensor issue we were having.
> 
> The code was there, so I thought it would be good to commit. I can also throw it away, for me tanto faz

So better keep. I found its use hard to follow at first glance, some doc would help with that, I think

**Review by wvmcastro — APPROVED** [review]

**Comment by doudou** [comment]
> some doc would help with that

True. Added more information to the usage message.
